### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by jdahm**
## Technical Description

Updates Dockerfile as part of the CMake modernization.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+RUN cmake -S .. -B . \
+    -DBUILD_TESTING=ON \
+    -DCMAKE_PREFIX_PATH=/usr/lib/llvm-9 \
+    -DCMAKE_INSTALL_PREFIX=/usr/local \
+    -GNinja
+RUN cmake --build . --target install -- -j$(nproc)
+
+FROM ubuntu:rolling AS dawn-exec
+LABEL Name=gtclang
+COPY --from=dawn-build /usr/local /usr/local
```
Comment (rheacangeo) [inline Dockerfile]: Interesting! This works, though strikes me as odd to copy the whole /usr/local... Alternatively you could try a 'FROM dawn-build as dawn-exec, and then you don't need to copy anything, you are building on top of that image. But I think either way is fine.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
+FROM ubuntu:rolling AS dawn-build
+RUN apt update && apt install -y \
+    build-essential ninja-build cmake \
+    python3 libpython-dev python3-setuptools \
+    llvm-9-dev libclang-9-dev \
+    libboost-dev && apt clean
```
Comment (rheacangeo) [inline Dockerfile]: Oh cool! I don't think I've used apt-clean at the end like this, presumably makes the image smaller!

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+**/.vs
+**/.vscode
+**/*.*proj.user
+**/*.dbmdl
+**/*.jfm
+**/azds.yaml
+**/bin
+**/charts
+**/docker-compose*
+**/Dockerfile*
```
Comment (rheacangeo) [inline .dockerignore]: I don't think the Dockerfile is typically part of the .dockerignore, but does no harm to have it here. Might as well keep things as small as possible!

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1,26 @@
+**/build
+**/install
+**/.classpath
+**/.dockerignore
```
Comment (rheacangeo) [inline .dockerignore]: docker ignore in the docker ignore. woah. meta :D.

Comment (jdahm) [reply .dockerignore]: Good catch! These were autogenerated by VSCode, but I can remove them.

File: Dockerfile (hunk 2 above)
Comment (jdahm) [reply Dockerfile]: Yah! I found that online somewhere. It takes a bit of time, but it does reduce the number of files in the image. Though, I'm not sure how large that is compared to another package dependency.

**Pull request merged**